### PR TITLE
Background zip

### PR DESCRIPTION
**Pull request by jbothma**

### CODE REVIEW

File: umibukela/background_tasks.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                'archive_filename': archive_filename,
+            })
+        ziptmpdir = mkdtemp()
+        try:
+            with NamedTemporaryFile(dir=ziptmpdir, delete=False) as zfh:
+                with ZipFile(zfh, 'w') as zf:
+                    for file in to_archive:
+                        zf.write(file['local_filename'], file['archive_filename'])
+                cycle = Cycle.objects.get(id=cycle_id)
+                cycle.materials.save('cycle-materials.zip', File(zfh))
```
Comment (longhotsummer) [inline umibukela/background_tasks.py:42]: Where is that saved in production?

File: umibukela/views.py [hunk 2]
```diff
@@ -750,3 +753,39 @@ def national_summary(request, survey_type_slug, cycle_id):
...
+                'dir': dir,
+            })
+        else:
+            url = request.build_absolute_uri(reverse('site-result-summary-pdf', kwargs=params))
+            artifacts.append({
+                'url': url,
+                'dir': dir,
+            })
+
+    background_tasks.create_zip(cycle_id, artifacts)
```
Comment (longhotsummer) [inline umibukela/views.py:790]: This runs it synchronously, right, not delayed?

Comment (jbothma) [reply umibukela/views.py:790]: the annotation on that function makes it just add it to the queue

Comment (longhotsummer) [reply umibukela/views.py:790]: Ah, I misread the docs, cool! Useful library, must remember it :)

File: umibukela/background_tasks.py (hunk 1 above)
Comment (jbothma) [reply umibukela/background_tasks.py:42]: the storage backend. it's a filefield on the cycle.

**Pull request merged**